features/steps/steps.py

```diff
@@ -186,6 +202,18 @@ def step_impl(context, field, values):
 
     context.applicable = getattr(context, 'applicable', True) and applicable
 
+
+@given('The {entity} is a part of another {other_entity} through the relationship {relationship}')
+def step_impl(context, entity, other_entity, relationship):
+    relationships = context.model.by_type(relationship)
+    instances = []
+    for rel in relationships:
+        if rel.RelatingObject.is_a(other_entity):
+            for obj in rel.RelatedObjects:
```

Comment:
See table above:

- not always named like that
- also not always an aggregate some relationships are 1..1 (see cardinality).

Reply:
Done, similarly to the above, that is using the lookup table. 
As for the point `also not always an aggregate some relationships are 1..1 (see cardinality)`: I've added a try/except block that checks, if the returned value is an iterator. If not, it "packages it" into a tuple, so the for loop would still work.